sample/src/main/java/com/rbrooks/indefinitepagerindicatorsample/viewPager2Sample/ViewPager2SampleFragment.kt

```diff
@@ -0,0 +1,92 @@
+import com.rbrooks.indefinitepagerindicatorsample.recyclerViewSample.PhotoItemRecyclerAdapter
+import com.rbrooks.indefinitepagerindicatorsample.util.OnPagerNumberChangeListener
+
+class ViewPager2SampleFragment : Fragment(), OnPagerNumberChangeListener, View.OnClickListener {
+
+    private var isVerticalEnabled = false
+
+    private var pagerAdapter: PhotoItemRecyclerAdapter? = null
+
+    private var _binding: FragmentViewPager2SampleBinding? = null
+    private val binding: FragmentViewPager2SampleBinding
+        get() = requireNotNull(_binding)
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+        _binding = FragmentViewPager2SampleBinding.inflate(inflater, container, false)
+
+        isVerticalEnabled = requireContext().getSharedPreferences(
+            MainActivity.SHARED_PREFERENCES,
+            androidx.appcompat.app.AppCompatActivity.MODE_PRIVATE
+        ).getBoolean(
+            MainActivity.isVerticalIndicatorKeyPreference,
+            false
+        )
+
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        setupViews()
+    }
+
+    private fun setupViews() {
+        pagerAdapter = PhotoItemRecyclerAdapter(requireContext())
+
+        with(binding) {
+            viewPager.adapter = pagerAdapter
+            if (isVerticalEnabled) {
+                verticalViewPagerIndicator.attachToViewPager2(viewPager)
+                verticalViewPagerIndicator.visibility = View.VISIBLE
+            } else {
+                horizontalViewPagerIndicator.attachToViewPager2(viewPager)
+                horizontalViewPagerIndicator.visibility = View.VISIBLE
+            }
+
+            previousButton.setOnClickListener(this@ViewPager2SampleFragment)
+            nextButton.setOnClickListener(this@ViewPager2SampleFragment)
+        }
+    }
+
+    override fun onPagerNumberChanged() {
+        pagerAdapter?.notifyDataSetChanged()
+    }
+
+    override fun onClick(v: View?) {
+        with(binding) {
+            val adapter = requireNotNull(viewPager.adapter)
+            when (v?.id) {
+                R.id.previous_button -> {
+                    if (viewPager.currentItem == 0) {
+                        viewPager.currentItem = adapter.itemCount - 1
+                    } else {
+                        viewPager.currentItem = viewPager.currentItem - 1
+                    }
+                }
+                R.id.next_button -> {
+                    if (viewPager.currentItem == adapter.itemCount - 1) {
+                        viewPager.currentItem = 0
+                    } else {
+                        viewPager.currentItem = viewPager.currentItem + 1
+                    }
+                }
+            }
+        }
+    }
+
+    override fun onDestroyView() {
+        super.onDestroyView()
```

Comment:
shouldn't we lose the references before calling onDestroyView?

Reply:
Afaik it doesn't really matter in what order you're doing it

Reply:
I think normally it's safer to get rid of the references before calling the super because you don't really know how the underlying references in the onDestroyView are being handled :) specially in Fragments since they tend to go through the whole lifecycle (onStop(), onPause(), onDestroy(), onResume()) more often.

```kotlin
override fun onDestroyView() {
     _binding = null
     pagerAdapter = null
     super.onDestroyView()
}
```

But that's just an opinion there, we can leave it as it is.